src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetCultureCommand.cs

```diff
@@ -1,23 +1,123 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Collections.Generic;
+using System.Globalization;
 using System.Management.Automation;
 
 namespace Microsoft.PowerShell.Commands
 {
     /// <summary>
-    /// Returns the thread's current culture.
+    /// Returns:
+    ///     - the thread's current culture
+    ///     - culture by name
+    ///     - list of all supported cultures.
     /// </summary>
-    [Cmdlet(VerbsCommon.Get, "Culture", HelpUri = "https://go.microsoft.com/fwlink/?LinkID=113312")]
+    [Cmdlet(VerbsCommon.Get, "Culture", DefaultParameterSetName = CurrentCultureParameterSet, HelpUri = "https://go.microsoft.com/fwlink/?LinkID=113312")]
     [OutputType(typeof(System.Globalization.CultureInfo))]
     public sealed class GetCultureCommand : PSCmdlet
     {
+        private const string CurrentCultureParameterSet = "CurrentCulture";
+        private const string NameParameterSet = "Name";
+        private const string ListAvailableParameterSet = "ListAvailable";
+
+        /// <summary>
+        /// Gets or sets culture names for which CultureInfo values are returned.
+        /// </summary>
+        [Parameter(ParameterSetName = NameParameterSet, Position = 0, ValueFromPipeline = true)]
+        [ValidateSet(typeof(ValidateCultureNamesGenerator))]
+        [ValidateNotNull]
+        public string[] Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets a switch to return current culture with user overrides (by default).
+        /// With the switch on, we return current culture without user overrides.
+        /// </summary>
+        [Parameter(ParameterSetName = CurrentCultureParameterSet)]
+        [Parameter(ParameterSetName = NameParameterSet)]
+        public SwitchParameter NoUserOverrides { get; set; }
+
+        /// <summary>
+        /// Gets or sets a switch to list all available cultures.
+        /// </summary>
+        [Parameter(ParameterSetName = ListAvailableParameterSet)]
+        public SwitchParameter ListAvailable { get; set; }
+
         /// <summary>
-        /// Output the current Culture info object.
+        /// Output:
+        ///     - the thread's current culture
+        ///     - culture by name
+        ///     - list of all supported cultures.
         /// </summary>
-        protected override void BeginProcessing()
+        protected override void ProcessRecord()
+        {
+            CultureInfo ci;
+
+            switch (ParameterSetName)
+            {
+                case CurrentCultureParameterSet:
+                    if (NoUserOverrides)
+                    {
+                        ci = CultureInfo.GetCultureInfo(Host.CurrentCulture.LCID);
```

Comment:
Please use `.Name` instead of `.LCID`, and we should probably do the same for all tests.

> **"LCIDs are being deprecated, and implementers are strongly encouraged to use locale names instead.** LCIDs can be used for backward compatibility, but as noted in section 2.2.1, there is no guarantee of LCID uniqueness when used with valid locale names not otherwise associated with an LCID. " - from "[MS-LCID]: Windows Language Code Identifier (LCID) Reference" - https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/MS-LCID/[MS-LCID].pdf

Reply:
I think the change is not related to and not critical for the PR. We can postpone the change until CoreFX obsolete the API.

Reply:
`[cultureinfo]::getcultures('allcultures') | group-object lcid | ? count -gt 1` will tell you that hundreds of predefined cultures share the same LCID, so this fix is an absolute must.

Reply:
Fixed.